drivers/platform/aducm3029/irq.c

```diff
@@ -0,0 +1,287 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "irq.h"
+#include "irq_extra.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/** The number of the first external interrupt, used by NVIC */
+#define BASE_XINT_NB			(XINT_EVT0_IRQn)
+
+/** Number of interrupts controllers available */
+#define NB_INTERRUPT_CONTROLLERS	1u
+
+/** Map the interrupt ID to the ADI_XINT_EVENT associated event */
+static const uint32_t id_map_event[NB_EXT_INTERRUPTS] = {
+	ADI_XINT_EVENT_INT0, // ID 0
+	ADI_XINT_EVENT_INT1, // ID 1
+	ADI_XINT_EVENT_INT2, // ID 2
+	ADI_XINT_EVENT_INT3  // ID 3
+};
+
+/******************************************************************************/
+/***************************** Global Variables *******************************/
+/******************************************************************************/
+
+/**
+ * Used to store the status of the driver. 1 if the driver is initialized, 0
+ * otherwise
+ */
+static uint32_t		initialized;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Call the user callback
+ * @param aducm_desc - Descriptor where the user callback is stared
+ * @param event - Event that generated the callback
+ * @param arg - Unused
+ */
+static void internal_callback(void *aducm_desc, uint32_t event, void *arg)
+{
+	struct aducm_irq_desc *desc = aducm_desc;
+
+	(void)arg;
+
+	if (event < NB_EXT_INTERRUPTS && desc->irq_handler[event])
+		desc->irq_handler[event]((void *)event);
+}
+
+/**
+ * @brief Initialized the controller for the ADuCM3029 external interrupts
+ *
+ * @param desc - Pointer where the configured instance is stored
+ * @param param - Configuration information for the instance
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t irq_ctrl_init(struct irq_ctrl_desc **desc,
+		      const struct irq_init_param *param)
+{
+	struct aducm_irq_desc		*aducm_desc;
+
+	if (!desc || !param || initialized)
+		return FAILURE;
+
+	*desc = calloc(1, sizeof(**desc));
```

Comment:
nitpick: we should be doing casts for calloc pointers from now on.

Reply:
![image](https://user-images.githubusercontent.com/16192750/74442682-96ba7480-4e7a-11ea-914f-f18daff2d9d2.png)

Reply:
Even considering this should we use the cast?

Reply:
Yes, because we are also trying to make the code compatible in a C++ environment too, and C++ compilers require explicit cast for void pointers to another type of pointer. Bumped into this one too.

Reply:
solve